app/models/user.rb

```diff
@@ -1,4 +1,16 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :recoverable,
-    :trackable, :validatable, :timeoutable
+    :trackable, :validatable, :timeoutable,
+    :omniauthable, omniauth_providers: [:mojsso]
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.provider = auth.provider
+      user.uid = auth.uid
+      user.email = auth.info.email
```

Comment:
Have you got a unique index on email? If so you could have the case that someone uses sso after having an account which will result in an error.